app/src/main/java/com/andela/art/login/LoginActivity.java

```diff
@@ -0,0 +1,175 @@
+package com.andela.art.login;
+
+import butterknife.BindView;
+import butterknife.ButterKnife;
+import butterknife.OnClick;
+
+import com.google.android.gms.auth.api.signin.GoogleSignInOptions;
+
+import android.app.ProgressDialog;
+import android.content.Context;
+import android.content.Intent;
+import android.content.pm.ActivityInfo;
+import android.support.annotation.NonNull;
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.util.Log;
+import android.view.Window;
+import android.view.WindowManager;
+import android.widget.Toast;
+
+import com.andela.art.serialentry.SerialEntryActivity;
+import com.andela.art.R;
+import com.google.android.gms.auth.api.signin.GoogleSignIn;
+import com.google.android.gms.auth.api.signin.GoogleSignInAccount;
+import com.google.android.gms.auth.api.signin.GoogleSignInClient;
+import com.google.android.gms.common.api.ApiException;
+import com.google.android.gms.tasks.OnCompleteListener;
+import com.google.android.gms.tasks.Task;
+import com.google.firebase.auth.AuthCredential;
+import com.google.firebase.auth.AuthResult;
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.auth.GoogleAuthProvider;
+import com.shobhitpuri.custombuttons.GoogleSignInButton;
+
+import uk.co.chrisjenx.calligraphy.CalligraphyConfig;
+import uk.co.chrisjenx.calligraphy.CalligraphyContextWrapper;
+
+
+public class LoginActivity extends AppCompatActivity implements LoginActivityMVP.LoginActivity {
+
+    @BindView(R.id.google_sign_in_button)
+    GoogleSignInButton button;
+    FirebaseAuth mAuth;
+    private final static int RC_SIGN_IN = 2;
+    GoogleSignInClient mGoogleSignInClient;
+    private static final String TAG = "LoginActivity";
+    FirebaseAuth.AuthStateListener mAuthListener;
+    // A progress dialog to display when the user is connecting in
+    // case there is a delay in any of the dialogs being ready.
+    private ProgressDialog mConnectionProgressDialog;
+
+    @Override
+    protected void onStart() {
+        super.onStart();
+        mAuth.addAuthStateListener(mAuthListener);
+    }
+
+    @Override
+    public void onStop() {
+        super.onStop();
+        if (mAuthListener != null) {
+            mAuth.removeAuthStateListener(mAuthListener);
+        }
+    }
+
+    @Override
+    protected void attachBaseContext(Context newBase) {
+        super.attachBaseContext(CalligraphyContextWrapper.wrap(newBase));
+    }
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        CalligraphyConfig.initDefault(new CalligraphyConfig.Builder()
```

Comment:
You can directly set the fonts on your XML file, I don't think you need this library. Just create a font folder in the res directory and use font family to set it. https://developer.android.com/guide/topics/ui/look-and-feel/fonts-in-xml.html

Reply:
No you don't need to bump up your API level. It's says in the documentation that `To use the Fonts in XML feature on devices running Android 4.1 (API level 16) and higher, use the Support Library 26.` Which you're already using

Reply:
Addressed.